### PR TITLE
[WIP] Docker plugin installation through CLI

### LINKED ISSUE #2320: Volume driver management
```
$ kontena volume driver add rexray/ebs:0.8.2 [--label region=eu-west-1] --kind block [name]
```

When user adds a driver to Kontena Master it is automatically distributed as a volume plugin to matching nodes (based on label filter).

Volume driver subcommand should have basic list/show/rm features.

### PR DESCRIPTION
**Pull request by jnummelin**
This PR makes installing Docker plugins on nodes possible through Kontena CLI.

## Questions

As each plugin (regardless of it's type) is installed exactly the same way, should the command be moved to be e.g. `kontena node plugin install ...`?

## TODO

- [ ] Collect each nodes result
- [ ] Make each node run installation in parallel
- [ ] Docs

fixes #2320

### CODE REVIEW

File: server/app/mutations/volumes/plugin_install.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+      end
+      string :label
+    end
+
+    def validate
+
+    end
+
+    def execute
+      # TODO
```
Comment (jakolehm) [inline server/app/mutations/volumes/plugin_install.rb]: Why are we sending rpc calls here? Imho more robust way is to pass plugin information to agent via agent pull (node info maybe?)

**Comment by jakolehm** [comment]
> As each plugin (regardless of it's type) is installed exactly the same way, should the command be moved to be e.g. kontena node plugin install ...?

No. It's not very convenient to run install command to every node... whole point of this install command is to automate things as much as possible. Meaning: you can run plugin install without any nodes, nodes will get matching plugins installed automatically when they join.

**Comment by jnummelin** [comment]
> No. It's not very convenient to run install command to every node
Regardless of the installation way (pull/push mode), I think it makes sense to have the command somewhere else than under `volumes`. Yes, volumes are the main use case for this but any other type of plugin is installed on the same way. So I thought the command could be under `kontena node plugins ...`. It's true that most of the sub-commands under `kontena node` are for single node, but would it be too bad to have even the pull mode installation as `kontena node plugin install ...`?

File: server/app/mutations/volumes/plugin_install.rb (hunk 1 above)
Comment (jnummelin) [inline server/app/mutations/volumes/plugin_install.rb]: Rcp push was more simple and straightforward. Yes, it has the drawback of installing plugins only on request and not automatically after a new node joins the grid.

**Comment by jakolehm** [comment]
Other `kontena node` commands are for single node, it's confusing to mix there something else. Maybe  `kontena grid plugin` would make more sense?

**Comment by SpComb** [comment]
Bumping WIP PR to 1.5, needs some major design refactoring (declarative pull vs RPC push).

**Comment by jnummelin** [comment]
rebased

**Pull request closed without merging**